iothub/device/src/Transport/AmqpTransportHandler.cs

```diff
@@ -987,7 +990,7 @@ private Twin TwinFromResponse(AmqpMessage message)
         {
             string path = this.BuildPath(CommonConstants.DeviceEventPathTemplate, CommonConstants.ModuleEventPathTemplate);
 
-            ReceivingAmqpLink messageReceivingLink = await this.IotHubConnection.CreateReceivingLinkAsync(path, this.iotHubConnectionString, this.deviceId, IotHubConnection.ReceivingLinkType.Events, this.prefetchCount, timeout, this.productInfo, cancellationToken);
+            ReceivingAmqpLink messageReceivingLink = await this.IotHubConnection.CreateReceivingLinkAsync(path, this.iotHubConnectionString, $"{this.deviceId}/{this.moduleId}", IotHubConnection.ReceivingLinkType.Events, this.prefetchCount, timeout, this.productInfo, cancellationToken);
```

Comment:
Can `this.ModuleId` be null (or empty string) here? Do you need to use something like the local method `BuildPath` for this?

Reply:
No, this is in module receive events code path.. so moduleId should never be null or empty.